samples/DataImporter/GraphEngine.DataImporter/FileReader.cs

```diff
@@ -27,17 +28,35 @@ internal static IEnumerable<string> ReadFile(string path)
             string filetype = Path.GetExtension(filename).ToLower();
 
             CompressionMode mode = CompressionMode.None;
-
+            long counter = 0;
             if (filetype == ".gz")
             {
                 mode = CompressionMode.GZip;
             }
             else if (filetype == ".zip")
             {
                 mode = CompressionMode.Zip;
-            }//TODO more
-
-            long counter = 0;
+                {
+                    using (ZipArchive archive = ZipFile.Open(path, ZipArchiveMode.Read))
```

Comment:
I suggest we regard a zip archive as a folder (we have the logic to scan a folder and obtain all entries). Once the entries in the zip archive are visible to the outside, the dispatcher can choose the right importer to process the data, thus removing the restrict of "same type".